node/src/reactor/main_reactor/validate.rs

```diff
@@ -25,9 +25,12 @@ impl MainReactor {
         rng: &mut NodeRng,
     ) -> ValidateInstruction {
         let queue_depth = self.contract_runtime.queue_depth();
-        if self.contract_runtime.queue_depth() > 0 {
-            debug!("Validate: should_validate queue_depth {}", queue_depth);
-            return ValidateInstruction::KeepUp;
+        if queue_depth > 0 {
+            warn!("Validate: should_validate queue_depth {}", queue_depth);
+            return ValidateInstruction::CheckLater(
```

Comment:
So basically this means that we prefer to continue to execute rather than try to keep-up if the queue is backed uo. 
But if this validator continues to be slow and it keeps lagging behind with more and more blocks, do we need an exit condition from here?

Reply:
it means it is preferable to remain in validate mode and allow time for the q to clear.